templatestore/frontend/src/components/TinyUrlComponent.js

```diff
@@ -0,0 +1,189 @@
+import React, { useState, useEffect } from 'react';
+import styles from '../style/templateScreen.less';
+import axios from 'axios';
+import { backendSettings } from './../utils.js';
+import TinyUrlDropdownComponent from './TinyUrlDropdownComponent';
+
+export default function TinyUrlComponent(props) {
+    if (props.contextData === '') return <></>;
+    var tinyUrlObjLocal = props.tinyUrlObj;
+    const [items, setItems] = useState([{ urlKey: "", expiry: "" }]);
+    const [visited, setVisited] = useState({});
+    useEffect(() => {populateItems(props.tinyUrlObj);}, [props.tinyUrlObj]);
+    function populateItems(tinyUrlObj) {
+        if (tinyUrlObj.length === 0) return;
+        let temp = [...items];
+        let tempvisited = { ...visited };
+        tinyUrlObj.forEach(obj => {
+            tempvisited[obj['urlKey']] = 1;
+            temp.splice(temp.length - 1, 0, {
```

Comment:
Conceptually this code is correct but it could be simplified using spread operator as:
temp = [...temp, {urlKey: <>, expiry:<>}]